### PR TITLE
Added Coding Resources and mock interview websites

### PR DESCRIPTION
**Pull request by Kritimakaushal**
Hey there!, Thanks for submitting a PR to this repo :smiley:  
Please complete the following checklist for your PR to be approved. Mark the items by adding an `x`.   
Example:  - [ ] Mark me as completed  
To mark: - [x] Mark me as completed

- [x] My Pull Request adheres to the **Contribution guidelines**.
- [x] I have added Competitive Coding Resources in the correct 'USEFUL_RESOURCES.md' file.
- [x] In `USEFUL_RESOURCES.md` I have provided a **correct link** to the resources.  
  (In case you are not adding a problem, you can freely mark this as complete)
- [x] The documentation part which I have given is my **own** and not plagiarized in any form.
- [x] The resources I have added are trusted by the public and indeed is very helpful for students preparation.
- [x] I have properly indented and **documented my resources and links**.
- [x] *I agree that all the above marked are true to the best of my knowledge and my PR can be closed if found otherwise.*

If any of the above is unmarked, please make the necessary changes and then mark them so that you're PR can be approved and merged.  
(Only completed checklist PRs will be reviewed)

### CODE REVIEW

File: USEFUL_RESOURCES.md [hunk 1]
```diff
@@ -7,6 +7,22 @@
 * [Codeforces](https://codeforces.com/)
 * [Hackerearth](https://www.hackerearth.com/challenges/)
 
+### [GeeksForGeeks](https://www.geeksforgeeks.org/) - Best place to work on all the theoritical subjects including Database Management System, Data Structures, Data Algorithms, Object Oriented Programming, Operating Systems and many more.
+
+#### Benefits -
```
Comment (Vishruth-S) [inline USEFUL_RESOURCES.md:12]: @Kritimakaushal  Thanks for contributing, but I think let's keep it simple, the `benefits` part seems unnecessary. Could you please remove that. The rest are fine ig

**Review by Vishruth-S — CHANGES_REQUESTED** [review]

File: USEFUL_RESOURCES.md (hunk 1 above)
Comment (Kritimakaushal) [reply USEFUL_RESOURCES.md:12]: Sure @Vishruth-S . Doing that.

Comment (Kritimakaushal) [reply USEFUL_RESOURCES.md:12]: So I need to make a pull request again right? @Vishruth-S

Comment (Kritimakaushal) [reply USEFUL_RESOURCES.md]: @Vishruth-S The updates are done.

**Review by Vishruth-S — APPROVED** [review]
@Kritimakaushal Thanks !😃 
Do star ⭐ this repo if you haven't already 🙃

**Pull request merged**